lib/datawarehouse/partitionRegistrar/index.ts

```diff
@@ -0,0 +1,87 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+import { CallbackFunction } from "@pulumi/aws/lambda";
+import { EventRuleEvent } from "@pulumi/aws/cloudwatch";
+import { getS3Location } from "../../utils";
+import { createPartitionDDLStatement } from "./partitionHelper";
+
+export class HourlyPartitionRegistrar extends pulumi.ComponentResource {
```

Comment:
Do we see a case for not having hourly buckets? Maybe Daily or every minute?

Reply:
Yeah I definitely can see some customization here for other use cases. For this one, we specifically care about hourly as that is all that kinesis firehose supports for the output folders. But it might make sense to refactor or create something new for registering the output for batch tables.